### PR TITLE
Submit s3 bucket script

### PR DESCRIPTION
**Pull request by schatten**
Summary
---------
List objects in an s3 bucket [with prefix] and submit all as signed urls to an HMA instance

Test Plan
---------
Screen Shot 2021-06-30 at 18 00 42

Verified that we got these images were processed on the cloudwatch dashboard.

### CODE REVIEW

File: hasher-matcher-actioner/scripts/script_utils.py [hunk 1]
```diff
@@ -127,6 +127,30 @@ def send_single_submission_url(
         )
         put_response.raise_for_status()
 
+    def submit_single_url(
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/scripts/script_utils.py:130]: `submit_from_url` instead?

until we batch the `single` is redundant anyway imo

File: hasher-matcher-actioner/scripts/submit_s3_bucket [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        "--prefix",
+        help="S3 prefix to narrow down which objects you want to submit.",
+        required=False,
+    )
+    parser.add_argument(
+        "--api_url",
+        help="The Base URL of API gateway that you want to submit to. eg. https://mc620fy2hf.execute-api.us-east-1.amazonaws.com/",
+        required=True,
+    )
+    parser.add_argument("--loop", action="store_true")
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/scripts/submit_s3_bucket:129]: nit: help text

File: hasher-matcher-actioner/scripts/submit_s3_bucket [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                    Params={
+                        "Bucket": self.bucket,
+                        "Key": object["Key"],
+                    },
+                    ExpiresIn=3 * 60 * 60,  # 3 hours from now.
+                )
+                callable(presigned_url)
+
+    def run(self):
+        if self.loop:
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/scripts/submit_s3_bucket:81]: Had to read through multiple times can you add a comment?

if loop=true does this just mean use multiple workers? Do they somehow share a  `paginator` how are repeats avoided?

**Review by BarrettOlson — APPROVED** [review]
I do think the script itself is a little hard to understand. Not sure I fully get 'loop.' but our scripts need consolidation and refactoring anyway so happy to stamp :)

File: hasher-matcher-actioner/scripts/submit_s3_bucket (hunk 3 above)
Comment (schatten) [reply hasher-matcher-actioner/scripts/submit_s3_bucket:81]: Added a comment. Repeats are not avoided. Loop merely causes the process to be repeated until Ctrl-Cd. We can't do 4000 QPS with 5000 images, so we loop and submit the same images again and again.

Comment (BarrettOlson) [reply hasher-matcher-actioner/scripts/submit_s3_bucket:81]: got it, I'd probably recommends renaming loop to storm then but fine either way.

**Pull request merged**